src/ceph-volume/ceph_volume/systemd/systemctl.py

```diff
@@ -34,6 +34,22 @@ def is_active(unit):
     )
     return rc == 0
 
+def get_running_osd_ids():
+    out, err, rc = process.call([
+        'systemctl',
+        'show',
+        '--no-pager',
+        '--property=Id',
+        '--state=running',
+        'ceph-osd@*',
+    ])
+    osd_ids = []
+    for line in out:
+        if line:
+            # example line looks like: Id=ceph-osd@1.service
+            osd_id = line.split("@")[1].split(".service")[0]
```

Comment:
Might bbe overkill, but are we *sure* that this will always return something like `ceph-osd@N.service` ? 

This type of parsing might break the whole scan if it gets an `IndexError` or a `TypeError`. I would add a check for non-zero exit status (looking at `rc`) and a try/except on this line

Reply:
> Might bbe overkill, but are we _sure_ that this will always return something like `ceph-osd@N.service` ?

I understand that concern. I've tested it on xenial and centos7 and it's there. I wanna say we control the name `ceph-osd@N.service` from our systemd units, but I'm not 100% positive. If you know of a more robust way to get the ID of running osds I'd be happy to implement it.

> 
> This type of parsing might break the whole scan if it gets an `IndexError` or a `TypeError`. I would add a check for non-zero exit status (looking at `rc`) and a try/except on this line

I can do that. Do you think we should fail hard if the rc is non-zero or just log a message and return an empty list?

Reply:
I think the latter... but not like an empty lis alwayst though, you have a couple of scenarios:

* non-zero exit status: means that whatever output is probably not OK to consume so we must return an empty list
* garbage that is not possible to parse: skip that item, log the warning, continue processing the rest of the output